src/instruction/binary_op.rs

```diff
@@ -172,7 +135,15 @@ impl TypeCheck for BinaryOp {
             return CheckedType::Unknown;
         }
 
-        l_type
+        match self.op {
+            Operator::Lt
+            | Operator::Gt
+            | Operator::LtEq
+            | Operator::GtEq
+            | Operator::Equals
+            | Operator::NotEquals => CheckedType::Resolved(TypeId::from("bool")),
+            _ => l_type,
+        }
```

Comment:
I think this might be missing checking that both the l_type and r_type are the same? Maybe I'm missing part of the function, I can't check it for now. If that is the case, could you add a test case for this as it is a typechecking error to operate on two types with different types and I should have added one already :)

Reply:
This done line 128. Don't know if this is tests though.

Reply:
Ah, great thanks for checking!